dnf5/context.cpp

```diff
@@ -634,7 +620,11 @@ bool Context::check_gpg_signatures(const libdnf::base::Transaction & transaction
 
 
 void Context::download_and_run(libdnf::base::Transaction & transaction) {
-    download_packages(transaction, nullptr);
+    try {
+        transaction.download();
+    } catch (const std::runtime_error & ex) {
+        std::cout << "Exception: " << ex.what() << std::endl;
+    }
```

Comment:
At this state you cannot continue with dnf5. The process must be terminated (nicely not with a core dump).

Reply:
There is a bug report for that - [Bug 2149377](https://bugzilla.redhat.com/show_bug.cgi?id=2149377) - [dnf5] dnf5 says not downloaded packages have invalid signatures

Reply:
Yeah, here I'm copying the existing error handling behavior from `download_packages`. But you're right, I'm not sure why the error is caught and logged there.

I will take this out.